motors/tacho_motor_class.c

```diff
@@ -721,16 +861,17 @@ static ssize_t speed_sp_store(struct device *dev, struct device_attribute *attr,
 			      const char *buf, size_t size)
 {
 	struct tacho_motor_device *tm = to_tacho_motor(dev);
-	int err, speed;
-
-	if (!(BIT(TM_SPEED_REGULATION_ON) & tm->ops->get_speed_regulations(tm->context)))
-		return -EOPNOTSUPP;
+	int err, speed, max_speed;
 
 	err = kstrtoint(buf, 10, &speed);
 	if (err < 0)
 		return err;
 
-	if (speed > 2000 || speed < -2000)
+	err = tm->ops->get_max_speed(tm->context, &max_speed);
+	if (err < 0)
+		return err;
+
+	if (abs(speed) > max_speed)
```

Comment:
Is 9V high enough? Many of the supported ev3dev platforms can have hight voltage power supplies (up to 16 or 18V probably although realistically, 12V max). The current max speeds are 1200 tacho counts/sec (200 rpm) for the large motor and 900 tacho counts per second (150 rpm) for the medium motor, which tells me I think we have them backwards - the medium motor turns faster. I'm looking at <http://www.philohome.com/motors/motorcomp.htm> and these are less than the 9V no-load value. We should repeat the experiment to verify, but philo's page is correct, the max values should be 1020 for NXT motor, 1050 for EV3 Large Motor and 1560 for EV3 Medium motor (assuming we are sticking with 9V no-load as max).